cli/format.go

```diff
@@ -355,8 +355,17 @@ func applyFormatters(ctx context.Context) func() error {
 			}
 
 			if len(matches) == 0 {
+				// exit with error if fail flag is set
+				if Cli.FailOnUnmatched {
+					return fmt.Errorf("no match found: %s", file.Path)
+				}
+
+				level := log.DebugLevel
+				if Cli.WarnUnmatched {
+					level = log.WarnLevel
+				}
```

Comment:
Looking at the branching here, how about `--on-unmatched=[debug|warn|fatal]`? Where `Cli.OnUnmatched` would store the log level.

Reply:
I took another stab bearing this in mind